cmd/event-notifier.go

```diff
@@ -230,7 +230,19 @@ func (en *eventNotifier) SendListenerEvent(arn string, event []NotificationEvent
 
 	ch, ok := en.internal.connectedListeners[arn]
 	if ok {
-		ch <- event
+		// To avoid blocking of send event over the channel
+		// while inside critical section.
+		// FIXME: The intermediary channel can be replaced by
+		// the actual http.ResponseWriter.
+		select {
+		case ch <- event:
+		default:
+			// When there is no receiver at the other end
+			// of the event channel, the channel blocks
```

Comment:
I don't think fix is correct - it would introduce a new problem. The ListenBucketNotificationHandler is the only go routine that listens on the channel and writes any received events to the http response writer. This means that between iterations of this loop, sending on the channel is blocked (because there is no listener). This in turn means that this select block will drop events, thus losing them.

See https://play.golang.org/p/oPmqFoaNA7 for a demonstration.

Reply:
Loosing is acceptable in this scenario @donatello , we are aware that it will happen. So there is a comment on why it is okay ignore. The reasoning is simple this only happens on a unbuffered channel when indeed there is no receiver. This part of the code is also scheduled to be quickly fixed by moving to response writers instead of channel.

Reply:
Also this situation only happens when there are other active listeners.. which are non-blocking so it is highly unlikely that all of them are blocking and not receiving. Single listener `default:` should never get executed.